src/ORM/Connect/Database.php

```diff
@@ -343,12 +343,8 @@ public function manipulate($manipulation)
                         $query->addWhere(array('"ID"' => $writeInfo['id']));
                     }
 
-                    // Test to see if this update query shouldn't, in fact, be an insert
-                    if ($query->toSelect()->count()) {
```

Comment:
This by itself isn't going to be enough. The insert logic in DataObject::write() initially writes to the base table to get the ID, and that relies on the base table insert being converted to an update here.

I thought the fix was to use affectedRows() to trigger the fallback to update?